### DISCUSSION TITLE
ESP32: default image enabled modules

### DISCUSSION THREAD

**Discussion by msperl** [arendst/Tasmota#11590] [Q&A] [answered]
At least in the developer firmware builds of esp32/tasmota32 there are "normal", "bluetooth" and a few other images prebuilt.

My question is: is this proliferation of images on devices with 4MB Flash really necessary?

The ESP32 chip comes with bluetooth, so why not include bluetooth with the default image?

The bluetooth image on the other hand does not come with shutter or berry support (and I have not looked into what else is missing)...

The sensor images have gone as a whole for the dev builds.

OK, it is easy to build your own image, but I wonder if it would not be better to come up with a common set of options that would fit 95% of all users without the need to custom rebuild?

**Comment by arendst** [comment, marked as answer]
Bluetooth has been removed from the "normal" builds because it had become very unstable. Recent changes made it more stable again so it may well be that in the near future bluetooth will be available in all builds.

Sensor build is obsolete as they are already included in the "normal" build.

**Reply by msperl** [reply]
great to hear that this is the intention!
